stdlib/tests/mem/mod.rs

```diff
@@ -0,0 +1,129 @@
+        program.kernel().clone(),
+        StackInputs::default(),
+        MemAdviceProvider::from(AdviceInputs::default()),
+    );
+    process.execute(&program).unwrap();
+
+    assert_eq!(process.get_memory_value(0, 1000), Some([ZERO, ZERO, ZERO, ONE]), "Address 1000");
+    assert_eq!(process.get_memory_value(0, 1001), Some([ZERO, ZERO, ONE, ZERO]), "Address 1001");
+    assert_eq!(process.get_memory_value(0, 1002), Some([ZERO, ZERO, ONE, ONE]), "Address 1002");
+    assert_eq!(process.get_memory_value(0, 1003), Some([ZERO, ONE, ZERO, ZERO]), "Address 1003");
+    assert_eq!(process.get_memory_value(0, 1004), Some([ZERO, ONE, ZERO, ONE]), "Address 1004");
+
+    assert_eq!(process.get_memory_value(0, 2000), Some([ZERO, ZERO, ZERO, ONE]), "Address 2000");
+    assert_eq!(process.get_memory_value(0, 2001), Some([ZERO, ZERO, ONE, ZERO]), "Address 2001");
+    assert_eq!(process.get_memory_value(0, 2002), Some([ZERO, ZERO, ONE, ONE]), "Address 2002");
+    assert_eq!(process.get_memory_value(0, 2003), Some([ZERO, ONE, ZERO, ZERO]), "Address 2003");
+    assert_eq!(process.get_memory_value(0, 2004), Some([ZERO, ONE, ZERO, ONE]), "Address 2004");
+}
+
+#[test]
+fn test_move_even_words_from_advice_stack_to_memory() {
+    let source = "
+    use.std::mem
+
+    begin
+        push.2          # number of words
+        push.1000       # target address
+        padw padw padw  # hasher state
+
+        exec.mem::move_even_words_from_advice_stack_to_memory
+    end
+    ";
+
+    let operand_stack = &[];
+    let advice_stack = &[1, 2, 3, 4, 5, 6, 7, 8];
+    let mut expected_stack =
+        stack_to_ints(&build_expected_perm(&[0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8]));
+    expected_stack.push(1002);
+    build_test!(source, operand_stack, &advice_stack).expect_stack(&expected_stack);
+}
+
+#[test]
+fn test_move_words_from_advice_stack_to_memory() {
+    let one_word = "
+    use.std::mem
+
+    begin
+        push.1000       # target address
+        push.1          # number of words
+
+        exec.mem::move_words_from_advice_stack_to_memory
+    end
+    ";
+
+    let operand_stack = &[];
+    let advice_stack = &[1, 2, 3, 4];
+    let mut expected_stack = stack_to_ints(&build_expected_hash(advice_stack));
+    expected_stack.push(1001);
+    build_test!(one_word, operand_stack, &advice_stack).expect_stack(&expected_stack);
+
+    let three_words = "
+    use.std::mem
+
+    begin
+        push.1000       # target address
+        push.3          # number of words
+
+        exec.mem::move_words_from_advice_stack_to_memory
+    end
+    ";
+
+    let operand_stack = &[];
+    let advice_stack = &[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    let mut expected_stack = stack_to_ints(&build_expected_hash(advice_stack));
+    expected_stack.push(1003);
+    build_test!(three_words, operand_stack, &advice_stack).expect_stack(&expected_stack);
+}
+
+#[test]
+fn test_move_words_from_advice_stack_to_memory_with_commitment() {
```

Comment:
For this test, should we also test the failure case (i.e., when the commitment doesn't match)?

Reply:
added a new test with an invalid commitment